### DISCUSSION TITLE
SSL error preventing JSON export—can you reproduce?

### DISCUSSION THREAD

**Discussion by ThatsJustCheesy** [Homebrew/discussions#3235] [Everyday usage]
I am getting an SSL error (from a cask vendor's cert expiring) that is preventing me from exporting JSON data for `--all`. Is anyone else able to reproduce this?

```
ian@Ian ~> brew info --json=v2 --all
Warning: Calling `Enumerable` methods on `Formula` is deprecated! Use `Formula.all` (but avoid looping over all formulae, it's slow and insecure) instead.
Error: Failure while executing; `/opt/homebrew/Library/Homebrew/shims/shared/curl --disable --cookie /dev/null
  --globoff --show-error --user-agent Homebrew/3.4.9-33-g3088bbf\ \(Macintosh\;\ arm64\ Mac\ OS\ X\ 12.3.1\)\ curl/7.79.1
  --header Accept-Language:\ en --retry 3 --fail --silent
  --location https://sequelpro.com/builds/latest-test-build.xml`
  exited with 60. Here's the output:
curl: (60) SSL certificate problem: certificate has expired
More details here: https://curl.se/docs/sslcerts.html

curl failed to verify the legitimacy of the server and therefore could not
establish a secure connection to it. To learn more about this situation and
how to fix it, please visit the web page mentioned above.
ian@Ian ~ [1|74]> 
```

(Hard wrapped manually for readability.)

<details>
<summary>`brew doctor` output (I don't there there's anything relevant; I put all those extra files there on purpose)</summary>
<pre><code>
ian@Ian ~> brew doctor
Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  lua@5.1
  php@7.4
  ruby@2.6
  sdl

Warning: "config" scripts exist outside your system or Homebrew directories.
`./configure` scripts often look for *-config scripts to determine if
software packages are installed, and which additional flags to use when
compiling and linking.

Having additional scripts in your path can confuse software installed via
Homebrew if the config script overrides a system or Homebrew-provided
script of the same name. We found the following "config" scripts:
  /opt/devkitpro/tools/bin/pkg-config
  /opt/devkitpro/tools/bin/x86_64-apple-darwin15-pkg-config

Warning: Unbrewed header files were found in /usr/local/include.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected header files:
  /usr/local/include/python3.7m/pygame/_camera.h
  /usr/local/include/python3.7m/pygame/_pygame.h
  /usr/local/include/python3.7m/pygame/_surface.h
  /usr/local/include/python3.7m/pygame/camera.h
  /usr/local/include/python3.7m/pygame/fastevents.h
  /usr/local/include/python3.7m/pygame/font.h
  /usr/local/include/python3.7m/pygame/freetype.h
  /usr/local/include/python3.7m/pygame/mask.h
  /usr/local/include/python3.7m/pygame/mixer.h
  /usr/local/include/python3.7m/pygame/palette.h
  /usr/local/include/python3.7m/pygame/pgarrinter.h
  /usr/local/include/python3.7m/pygame/pgbufferproxy.h
  /usr/local/include/python3.7m/pygame/pgcompat.h
  /usr/local/include/python3.7m/pygame/pgimport.h
  /usr/local/include/python3.7m/pygame/pgopengl.h
  /usr/local/include/python3.7m/pygame/pgplatform.h
  /usr/local/include/python3.7m/pygame/pygame.h
  /usr/local/include/python3.7m/pygame/scrap.h
  /usr/local/include/python3.7m/pygame/surface.h
  /usr/local/include/rpatk/rex/rexcompiler.h
  /usr/local/include/rpatk/rex/rexdb.h
  /usr/local/include/rpatk/rex/rexdef.h
  /usr/local/include/rpatk/rex/rexdfa.h
  /usr/local/include/rpatk/rex/rexdfaconv.h
  /usr/local/include/rpatk/rex/rexdfasimulator.h
  /usr/local/include/rpatk/rex/rexdfatypes.h
  /usr/local/include/rpatk/rex/rexfragment.h
  /usr/local/include/rpatk/rex/rexnfasimulator.h
  /usr/local/include/rpatk/rex/rexregex.h
  /usr/local/include/rpatk/rex/rexstate.h
  /usr/local/include/rpatk/rex/rextransition.h
  /usr/local/include/rpatk/rlib/rarray.h
  /usr/local/include/rpatk/rlib/ratomic.h
  /usr/local/include/rpatk/rlib/rbuffer.h
  /usr/local/include/rpatk/rlib/rcarray.h
  /usr/local/include/rpatk/rlib/rcharconv.h
  /usr/local/include/rpatk/rlib/rgc.h
  /usr/local/include/rpatk/rlib/rharray.h
  /usr/local/include/rpatk/rlib/rhash.h
  /usr/local/include/rpatk/rlib/rlist.h
  /usr/local/include/rpatk/rlib/rmap.h
  /usr/local/include/rpatk/rlib/rmath.h
  /usr/local/include/rpatk/rlib/rmem.h
  /usr/local/include/rpatk/rlib/robject.h
  /usr/local/include/rpatk/rlib/rref.h
  /usr/local/include/rpatk/rlib/rspinlock.h
  /usr/local/include/rpatk/rlib/rstring.h
  /usr/local/include/rpatk/rlib/rutf.h
  /usr/local/include/rpatk/rpa/rpa.h
  /usr/local/include/rpatk/rpa/rpabitmap.h
  /usr/local/include/rpatk/rpa/rpacache.h
  /usr/local/include/rpatk/rpa/rpacompiler.h
  /usr/local/include/rpatk/rpa/rpadbex.h
  /usr/local/include/rpatk/rpa/rpadbexpriv.h
  /usr/local/include/rpatk/rpa/rpaerror.h
  /usr/local/include/rpatk/rpa/rpaoptimization.h
  /usr/local/include/rpatk/rpa/rpaparser.h
  /usr/local/include/rpatk/rpa/rparecord.h
  /usr/local/include/rpatk/rpa/rpastat.h
  /usr/local/include/rpatk/rpa/rpastatpriv.h
  /usr/local/include/rpatk/rpa/rpavm.h
  /usr/local/include/rpatk/rtypes.h
  /usr/local/include/rpatk/rvm/rvmcodegen.h
  /usr/local/include/rpatk/rvm/rvmcodemap.h
  /usr/local/include/rpatk/rvm/rvmcpu.h
  /usr/local/include/rpatk/rvm/rvmerror.h
  /usr/local/include/rpatk/rvm/rvmreg.h
  /usr/local/include/rpatk/rvm/rvmrelocmap.h
  /usr/local/include/rpatk/rvm/rvmscope.h

Warning: Unbrewed '.pc' files were found in /usr/local/lib/pkgconfig.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected '.pc' files:
  /usr/local/lib/pkgconfig/cllvm.pc

Warning: Unbrewed static libraries were found in /usr/local/lib.
If you didn't put them there on purpose they could cause problems when
building Homebrew formulae, and may need to be deleted.

Unexpected static libraries:
  /usr/local/lib/libglfw3.a
  /usr/local/lib/librex.a
  /usr/local/lib/librlib.a
  /usr/local/lib/librpa.a
  /usr/local/lib/librvm.a
  /usr/local/lib/oogl.a
  /usr/local/lib/tjcengine-2.a
  /usr/local/lib/tjcengine.a
</code></pre>
</details>

Thanks!

**Comment by SMillerDev** [comment]
I get the same error and it prevents me from exporting this too. 
```console
$ brew info --json=v2 --all
> Error: Calling `Enumerable` methods on `Formula` is deprecated! Use `Formula.all` (but avoid looping over all formulae, it's slow and insecure) instead.
```

**Comment by ThatsJustCheesy** [comment]
I see the deprecation error was fixed—many thanks!

The expired cert error has also gone away for me (though the vendor still hasn't renewed their cert). I'm not sure if it was fixed in Homebrew core, or in homebrew-casks, or if it was fixed indirectly by using `Formula.all`. But regardless, it works now :)
